### PR TITLE
feat(Provider): set scrollbar styles

### PR DESCRIPTION
**Pull request by layershifter**
This adds styles for scrollbar to Teams theme. I defined styles in `ProviderBox` instead of `staticStyles` to avoid collisions.

### CODE REVIEW

File: packages/react/src/lib/handleRef.ts [hunk 1]
```diff
@@ -21,9 +21,8 @@ const handleRef = <N>(ref: React.Ref<N>, node: N) => {
   }
 
   if (ref !== null && typeof ref === 'object') {
-    // @ts-ignore The `current` property is defined as readonly, however it's a valid way because
-    // `ref` is a mutable object
-    ref.current = node
+    // The `current` property is defined as readonly, however it's a valid way because `ref` is a mutable object
+    ;(ref as React.MutableRefObject<N>).current = node
```
Comment (mnajdova) [inline packages/react/src/lib/handleRef.ts:25]: Is the `;` cruft?

File: packages/react/src/themes/base/components/Provider/providerBoxVariables.ts [hunk 2]
```diff
@@ -1,9 +1,31 @@
...
 }
 
 export default (siteVariables): ProviderBoxVariables => ({
   background: siteVariables.bodyBackground,
   color: siteVariables.bodyColor,
+
+  scrollbarHeight: pxToRem(16),
+  scrollbarWidth: pxToRem(16),
+
+  scrollbarThumbBackgroundColor: siteVariables.gray06,
```
Comment (mnajdova) [inline packages/react/src/themes/base/components/Provider/providerBoxVariables.ts]: Oops, base theme doesn't have `gray06` and `gray04` :S

File: packages/react/src/lib/handleRef.ts (hunk 1 above)
Comment (layershifter) [reply packages/react/src/lib/handleRef.ts:25]: Nope and this is sad: https://github.com/prettier/prettier/issues/4630 😭

File: packages/react/src/themes/base/components/Provider/providerBoxVariables.ts (hunk 2 above)
Comment (layershifter) [reply packages/react/src/themes/base/components/Provider/providerBoxVariables.ts]: Definitely, I initially overlooked the path of styles, now they are in correct place.

P.S. Motivation to add them only to Teams theme:
- one of the frequent questions in SUI: "How I can disable custom scrollbars?"
- this feature is not cross-browser

File: packages/react/src/lib/handleRef.ts (hunk 1 above)
Comment (layershifter) [reply packages/react/src/lib/handleRef.ts:25]: image
https://stardust-ui.github.io/react/shorthand-props
The same case, actually 😨

Comment (mnajdova) [reply packages/react/src/lib/handleRef.ts:25]: Ok, just to mention I hate this, especially that we are not using `;` at all... Sad day for me :D

File: packages/react/src/themes/base/components/Provider/providerBoxVariables.ts (hunk 2 above)
Comment (mnajdova) [reply packages/react/src/themes/base/components/Provider/providerBoxVariables.ts]: I would really love to see this as part of base theme, but I see your points. Let's think about this in the future. The base theme's provider should have custom scrollbar by default, at least in my opinion. Agreed that we should have way of enabling or disabling this behavior as well. The main reason for this is that, I don't see much different ways of defining the scrollbar, more over we can customize it trough it's variables.

File: packages/react/src/lib/handleRef.ts (hunk 1 above)
Comment (mnajdova) [reply packages/react/src/lib/handleRef.ts:25]: Even worse with the provided example, it is used in our guides... :\

**Review by mnajdova — APPROVED** [review]

**Review by miroslavstastny — COMMENTED** [review]
Changes look good, but are not covered by any test. Would it make sense to add a screener test for that?

**Pull request merged**